src/store/modules/orders/actions.ts

```diff
@@ -0,0 +1,192 @@
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+
+    try {
+      resp = await OrderService.getPackedOrders(payload)
+      if (resp.status === 200 && resp.data.count > 0 && !hasError(resp)) {
+        let packedOrders = resp.data.docs;
+        const total = resp.data.count;
+        if(payload.viewIndex && payload.viewIndex > 0) packedOrders = state.packedOrders.list.concat(packedOrders)
+        commit(types.ORDERS_PACKED_INITIAL, { packedOrders, total })
+        if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+      } else {
+        showToast(translate("Orders Not Found"))
+      }
+      if (payload.viewIndex === 0) emitter.emit("dismissLoader");
+    } catch(err) {
+      console.log(err)
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  async deliverShipment ({ dispatch }, order) {
+    emitter.emit("presentLoader");
+
+    const params = {
+      shipmentId: order.shipmentId,
+      statusId: 'SHIPMENT_SHIPPED'
+    }
+
+    let resp;
+
+    try {
+      resp = await OrderService.updateShipment(params)
+      if (resp.status === 200 && !hasError(resp)) {
+        showToast(translate(`Order delivered to ${order.customerName}`))
+        emitter.emit("dismissLoader");
+      } else {
+        emitter.emit("dismissLoader");
+        showToast(translate("Something went wrong"))
+      }
+    } catch(err) {
+      console.log(err)
+      emitter.emit("dismissLoader");
+      showToast(translate("Something went wrong"))
+    }
+
+    return resp;
+  },
+
+  getShipmentMethod({ commit }, payload) {
+    /* To display the button label as per the shipmentMethodTypeId, this will only used on orders segment.
+        Because we get the shipmentMethodTypeId on items level in wms-orders API.
+        As we already get shipmentMethodTypeId on order level in readytoshiporders API hence we will not use this method on packed orders segment.
+    */
+    return payload.items.find((ele: any) => ele.shipGroupSeqId == payload.shipGroupSeqId).shipmentMethodTypeId
+  },
+
+  async packDeliveryItems ({ commit }, shipmentId) {
+    const params = {
+      shipmentId: shipmentId,
+      statusId: 'SHIPMENT_PACKED'
+    }
+    return await OrderService.updateShipment(params)
+  },
+
+  async quickShipEntireShipGroup ({ dispatch }, payload) {
+    emitter.emit("presentLoader")
+
+    const params = {
+      orderId: payload.order.orderId,
+      setPackedOnly: 'Y',
+      dimensionUomId: 'WT_kg',
+      shipmentBoxTypeId: 'YOURPACKNG',
+      weight: '1',
+      weightUomId: 'WT_kg',
+      facilityId: payload.facilityId,
+      shipGroupSeqId: payload.shipGroup
+    }
+    
+    let resp;
+
+    try {
+      resp = await OrderService.quickShipEntireShipGroup(params)
+      if (resp.status === 200 && !hasError(resp) && resp.data._EVENT_MESSAGE_) {
+        const shipmentMethod = await dispatch('getShipmentMethod', {shipGroupSeqId: payload.shipGroup, items: payload.order.items})
```

Comment:
I think we should use the state directly. The action is not needed at all.

Reply:
Removed the state definition and used directly the state.